src/EditorFeatures/Test2/IntelliSense/VisualBasicCompletionCommandHandlerTests.vb

```diff
@@ -496,6 +495,42 @@ End Class
             End Using
         End Function
 
+        <MemberData(NameOf(AllCompletionImplementations))>
+        <WpfTheory, Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function TestNavigationOutBeforeCompletedComputation(completionImplementation As CompletionImplementation) As Task
+            ' Simulate a very slow completionImplementation provider.
+            Dim e = New ManualResetEvent(False)
+            Dim provider = CreateTriggeredCompletionProvider(e)
+
+            Using state = TestStateFactory.CreateVisualBasicTestState(completionImplementation,
+                              <document>
+                                Class Program
+                                    Shared Sub Main(args As String())
+                                        Program$$
+                                    End Sub
+                                End Class
+                              </document>, extraCompletionProviders:={provider})
+
+                Await state.AssertNoCompletionSession()
+                state.SendTypeChars(".Ma")
+
+                ' We should not have a session now.  Note: do not block as this will just hang things
+                ' since the provider will not return.
+                state.AssertNoCompletionSessionWithNoBlock()
+
+                ' Now, navigate using the caret.
+                state.SendDownKey()
+
+                ' allow the provider to continue
+                e.Set()
+
+                ' Caret was intended to be moved out of the span. 
+                ' Therefore, we should cancel the completion And move the caret.
+                Await state.AssertNoCompletionSession()
+                ' We do not check for the current line because the test infrastructure does not handle moving the caret down.
```

Comment:
this part seems a pity.  it would be really good to have validation that this was treated as a move.

Reply:
Thank you, @CyrusNajmabadi ! I have added some mock of up/down keys to the test infrastructure.